pyglet/text/caret.py

```diff
@@ -110,6 +116,8 @@ def __init__(self, layout: IncrementalTextLayout, batch: Batch | None = None,
 
         self.visible = True
 
+        self._window = window
+        self.push_handlers(window)
```

Comment:
No longer need to push the window as a handler if the Caret is dispatching. (Also the value could be None, so this would fail regardless)